apps/angular/crud/src/app/store/tasks/task.effects.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { catchError, exhaustMap, map, of, switchMap } from 'rxjs';
+import { TaskService } from '../../service/task.service';
+import { appActions } from '../commons/app.action';
+import { taskActions } from './task.action';
+
+@Injectable()
+export class TasksEffects {
+  constructor(
+    private actions$: Actions,
+    private service: TaskService,
+  ) {}
+
+  loadTasksEffect = createEffect(() =>
+    this.actions$.pipe(
+      ofType(taskActions.loadTasksAction),
+      exhaustMap((action) => {
+        return this.service.getAllData().pipe(
+          map((data) => {
+            return taskActions.loadTasksSuccess({ list: data });
+          }),
+          catchError((error) =>
+            of(
+              appActions.showAlert({
+                message: 'Fail fetching data:: ' + error.message,
+                resultType: 'fail',
+              }),
+            ),
+          ),
+        );
+      }),
+    ),
+  );
+
+  updTaskEffect = createEffect(() =>
+    this.actions$.pipe(
+      ofType(taskActions.updTaskAction),
+      switchMap((action) => {
+        return this.service.putEntity(action.task).pipe(
+          switchMap((data) => {
+            return of(
+              taskActions.updTaskSuccess({ task: action.task }),
+              appActions.showAlert({
+                message: 'Task updated',
+                resultType: 'pass',
+              }),
+            );
+          }),
+          catchError((error) =>
+            of(
+              appActions.showAlert({
+                message: 'Fail updating data:: ' + error.message,
+                resultType: 'fail',
+              }),
+            ),
+          ),
+        );
+      }),
+    ),
+  );
+
+  delTaskEffect = createEffect(() =>
+    this.actions$.pipe(
+      ofType(taskActions.delTaskAction),
+      switchMap((action) => {
```

Comment:
be careful with switchMap on delete, post, put. 
You will only cancel the request, not the change in the database.

Reply:
Strongly important, thank you!